test/common/common/utility_test.cc

```diff
@@ -118,6 +118,12 @@ TEST(StringUtil, split) {
   EXPECT_EQ(std::vector<std::string>{"hello"}, StringUtil::split("hello", ""));
 }
 
+TEST(StringUtil, join) {
+  EXPECT_EQ("hello,world", StringUtil::join(std::vector<std::string>{"hello", "world"}, ","));
```

Comment:
added that in https://github.com/lyft/envoy/pull/1018/commits/9dcbefe68cae44165d24c9ea000e8d780127b571.